### PR TITLE
New Adapter: Edge226

### PR DESCRIPTION
**Pull request by Edge226Ads**
New adapter Edge226
Dev docs pull request: https://github.com/prebid/prebid.github.io/pull/4864

### CODE REVIEW

File: static/bidder-info/edge226.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+endpoint: "http://ssp.dauup.com/pserver"
+maintainer:
+  email: "audit@edge226.com "
```
Comment (Sonali-More-Xandr) [inline static/bidder-info/edge226.yaml:3]: @Edge226Ads We have sent a verification mail to `audit@edge226.com`. Please reply to that mail with "Received" to complete the verification process.
Thank you!

Comment (Sonali-More-Xandr) [inline static/bidder-info/edge226.yaml:3]: Please remove the extra space after `"audit@edge226.com "`. It should be `"audit@edge226.com"`

Comment (Sonali-More-Xandr) [reply static/bidder-info/edge226.yaml:3]: Verification completed!

File: static/bidder-info/edge226.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    - native
+    
+  app:
+    mediaTypes:
+    - banner
+    - video
+    - native
+userSync:
+  redirect:
+    url: "https://ssp.dauup.com/pserver?gdpr={{.GDPR}}&gdpr_consent={{.GDPRConsent}}&us_privacy={{.USPrivacy}}&redir={{.RedirectURL}}"
```
Comment (Sonali-More-Xandr) [inline static/bidder-info/edge226.yaml:19]: The redirect url is not working for me
https://ssp.dauup.com/pserver?gdpr=&gdpr_consent=&us_privacy=&redir=https%3A%2F%2Fib.adnxs.com%2Fsetuid%3Fbidder%3Dedge226%26gdpr%3D%26gdpr_consent%3D%26gpp%3D%26gpp_sid%3D%26f%3Db%26uid%3D%5BUID%5D

image

File: adapters/edge226/edge226.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+			bidResponse.Bids = append(bidResponse.Bids, b)
+		}
+	}
+	return bidResponse, nil
+}
+
+func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
+	for _, imp := range imps {
+		if imp.ID == impID {
+			if imp.Banner != nil {
```
[the remark this answers is not in the document]
Comment (gargcreation1992) [reply adapters/edge226/edge226.go:139]: @Edge226Ads Requestin you to please address the above comment.

File: adapters/edge226/edge226.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
```
Comment (gargcreation1992) [inline adapters/edge226/edge226.go:110]: This is correct. But you could also use - 
```
if adapters.IsResponseStatusCodeNoContent(responseData) {
     return nil, nil
}

if err := adapters.CheckResponseStatusCodeForErrors(responseData); err != nil {
     return nil, []error{err}
}
	```

File: adapters/edge226/edge226.go [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+				Bid:     &seatBid.Bid[i],
+				BidType: bidType,
+			}
+			bidResponse.Bids = append(bidResponse.Bids, b)
+		}
+	}
+	return bidResponse, nil
+}
+
+func getMediaTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
```
Comment (gargcreation1992) [inline adapters/edge226/edge226.go:136]: Please address the below comment. You should be setting https://github.com/prebid/openrtb/blob/main/openrtb2/bid.go#L334 in the bidder response.

File: static/bidder-info/edge226.yaml [hunk 6]
```diff
@@ -0,0 +1,16 @@
+endpoint: "http://ssp.dauup.com/pserver"
+maintainer:
+  email: "audit@edge226.com"
```
Comment (gargcreation1992) [inline static/bidder-info/edge226.yaml:3]: Sent a verification email. Please reply with "received"

**Review by gargcreation1992 — COMMENTED** [review]
Few suggestions

File: adapters/edge226/edge226.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Sonali-More-Xandr) [reply adapters/edge226/edge226.go:139]: @Edge226Ads Please share your thoughts on the above comment.

File: static/bidder-info/edge226.yaml (hunk 6 above)
Comment (Sonali-More-Xandr) [reply static/bidder-info/edge226.yaml:3]: Received response. Email verification completed!

File: static/bidder-info/edge226.yaml (hunk 2 above)
Comment (gargcreation1992) [reply static/bidder-info/edge226.yaml]: @Edge226Ads The cookie sync url is still failing with 404. Please re-verify.

**Comment by gargcreation1992** [comment]
@Edge226Ads User sync url is not working. Requesting you to please address the comment.

**Comment by Edge226Ads** [comment]
@gargcreation1992 Sync removed https://github.com/prebid/prebid-server/pull/3105/commits/0df50dbfde17d5372bf25bd23dcef0b5243f743b

**Review by gargcreation1992 — APPROVED** [review]

**Pull request merged**